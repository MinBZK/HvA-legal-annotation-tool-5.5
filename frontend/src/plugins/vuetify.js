// /**
//  * plugins/vuetify.js
//  *
//  * Framework documentation: https://vuetifyjs.com`
//  */
//
// // Styles
// import '@mdi/font/css/materialdesignicons.css'
// import 'vuetify/styles'
// import Vue from 'vue';
// import Vuetify from 'vuetify/lib/framework';
//
// Vue.use(Vuetify);
//
// // Composables
// import { createVuetify } from 'vuetify'
//
// // https://vuetifyjs.com/en/introduction/why-vuetify/#feature-guides
// export default createVuetify({
//   theme: {
//     themes: {
//       light: {
//         colors: {
//           primary: '#1867C0',
//           secondary: '#5CBBF6',
//         },
//       },
//     },
//   },
// })
// plugins/vuetify.js

// Remove this import line
// import Vue from 'vue';

// Add these import lines
import { createApp } from 'vue';
import Vuetify from 'vuetify/lib/framework';

const app = createApp();
app.use(Vuetify);

// Styles
import '@mdi/font/css/materialdesignicons.css'
import 'vuetify/styles'

// Composables
import { createVuetify } from 'vuetify'

// https://vuetifyjs.com/en/introduction/why-vuetify/#feature-guides
export default createVuetify({
  theme: {
    themes: {
      light: {
        colors: {
          primary: '#1867C0',
          secondary: '#5CBBF6',
        },
      },
    },
  },
})
