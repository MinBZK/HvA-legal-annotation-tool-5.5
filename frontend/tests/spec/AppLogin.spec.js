import { expect } from 'vitest';
import { mount } from '@vue/test-utils';
import LoginComponent from '@/components/AppLogin.vue';
import { createVuetify } from 'vuetify';

import { createPinia, setActivePinia } from 'pinia';


// Stub XMLSerializer to prevent interference with serialization
class StubbedXMLSerializer {
  serializeToString() {
    return '';
  }
}

global.XMLSerializer = StubbedXMLSerializer;

// Before each test
beforeEach(() => {
  const pinia = createPinia();
  setActivePinia(pinia);
});

const vuetify = createVuetify();

test('renders login component correctly', () => {

  const wrapper = mount(LoginComponent, {
    global: {
      plugins: [vuetify, createPinia()],
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('initializes with correct data', () => {

  const wrapper = mount(LoginComponent, {
    global: {
      plugins: [vuetify, createPinia()],
    },
  });

  expect(wrapper.vm.username).toBe('');
  expect(wrapper.vm.password).toBe('');
  expect(wrapper.vm.loading).toBe(false);
  expect(wrapper.vm.errors.username).toEqual([]);
  expect(wrapper.vm.errors.password).toEqual([]);
  expect(wrapper.vm.loginError).toBe(false);
  expect(wrapper.vm.loginErrorMessage).toBe('');
  expect(wrapper.vm.loginSuccess).toBe(false);
  expect(wrapper.vm.loginSuccessMessage).toBe('');
});

test('validates form inputs correctly', async () => {

  const wrapper = mount(LoginComponent, {
    global: {
      plugins: [vuetify, createPinia()],
    },
  });

  await wrapper.setData({
    username: '',
    password: '',
  });

  await wrapper.find('form').trigger('submit.prevent');
  await wrapper.vm.$nextTick();

  expect(wrapper.vm.errors.username).toContain('Gebruikersnaam is verplicht.');
  expect(wrapper.vm.errors.password).toContain('Wachtwoord is verplicht.');
});

test('updates data on user input', async () => {

  const wrapper = mount(LoginComponent, {
    global: {
      plugins: [vuetify, createPinia()],
    },
  });

  // Simulate user input
  await wrapper.setData({
    username: 'testuser',
    password: 'testpassword',
  });

  // Assert that the component's data is updated
  expect(wrapper.vm.username).toBe('testuser');
  expect(wrapper.vm.password).toBe('testpassword');
});







