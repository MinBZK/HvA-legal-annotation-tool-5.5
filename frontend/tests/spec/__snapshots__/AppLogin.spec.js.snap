// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders login component correctly 1`] = `
"<div class="v-container v-container--fluid v-locale--is-ltr">
  <div class="v-row justify-center">
    <div class="v-col-sm-8 v-col-md-4 v-col-12">
      <div class="v-card v-theme--light v-card--density-default v-card--variant-elevated elevation-12">
        <!---->
        <div class="v-card__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background" style="width: 100%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class="v-card-title text-center">
          <h2>Inloggen</h2>
        </div>
        <div class="v-card-text">
          <!-- Form for user login -->
          <form class="v-form" novalidate="">
            <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field mb-4">
              <!---->
              <div class="v-input__control">
                <div class="v-field v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
                  <div class="v-field__overlay"></div>
                  <div class="v-field__loader">
                    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                      <!---->
                      <div class="v-progress-linear__background" style="width: 100%;"></div>
                      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                        <div class="v-progress-linear__indeterminate">
                          <div class="v-progress-linear__indeterminate long"></div>
                          <div class="v-progress-linear__indeterminate short"></div>
                        </div>
                      </transition-stub>
                      <!---->
                    </div>
                  </div>
                  <!---->
                  <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
                      <!---->Gebruikersnaam
                    </label><label class="v-label v-field-label" for="input-0">
                      <!---->Gebruikersnaam
                    </label>
                    <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" outlined="" required="" class="v-field__input">
                    <!---->
                  </div>
                  <!---->
                  <!---->
                  <div class="v-field__outline">
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <div class="v-input__details">
                <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
                  <!---->
                </transition-group-stub>
                <!---->
              </div>
            </div>
            <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field mb-4">
              <!---->
              <div class="v-input__control">
                <div class="v-field v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
                  <div class="v-field__overlay"></div>
                  <div class="v-field__loader">
                    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                      <!---->
                      <div class="v-progress-linear__background" style="width: 100%;"></div>
                      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                        <div class="v-progress-linear__indeterminate">
                          <div class="v-progress-linear__indeterminate long"></div>
                          <div class="v-progress-linear__indeterminate short"></div>
                        </div>
                      </transition-stub>
                      <!---->
                    </div>
                  </div>
                  <!---->
                  <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-2">
                      <!---->Wachtwoord
                    </label><label class="v-label v-field-label" for="input-2">
                      <!---->Wachtwoord
                    </label>
                    <!----><input size="1" type="password" id="input-2" aria-describedby="input-2-messages" outlined="" required="" class="v-field__input">
                    <!---->
                  </div>
                  <!---->
                  <!---->
                  <div class="v-field__outline">
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <div class="v-input__details">
                <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-2-messages">
                  <!---->
                </transition-group-stub>
                <!---->
              </div>
            </div><!-- Button for login submission --><button type="submit" class="v-btn v-btn--block v-btn--elevated v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated white--text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator="">Inloggen</span>
              <!---->
              <!---->
            </button>
            <div class="v-row mt-4">
              <!-- Other content -->
            </div>
          </form><!-- Display error messages -->
          <!--v-if-->
          <!-- Display success message -->
          <!--v-if-->
        </div>
        <!---->
        <!----><span class="v-card__underlay"></span>
      </div>
    </div>
  </div>
</div>"
`;
